node/channel.js

```diff
@@ -319,6 +319,11 @@ TChannel.prototype.listen = function listen(port, host, callback) {
 TChannel.prototype.register = function register(name, options, handler) {
     var self = this;
 
+    // If we are the top channel
+    if (!self.topChannel) {
```

Comment:
I'd rather seen this done by quack than hack:
- check if the handler can register, if so register on it
- otherwise it's an error